tsdb/chunks/head_chunks_test.go

```diff
@@ -231,15 +237,18 @@ func TestChunkDiskMapper_Truncate(t *testing.T) {
 
 	// Truncating files after restart.
 	require.NoError(t, hrw.Truncate(sixthFileMinT))
-	verifyFiles([]int{6, 7, 8, 9, 10})
 
-	// As the last file was empty, this creates no new files.
```

Comment:
Since this cannot be tested easily now, we can now test if calling 2 Truncate() back to back creates only 1 file. So we can call `require.NoError(t, hrw.Truncate(sixthFileMinT+1))` additionally here before adding the chunk. And as I said in another comment, we check here again that there is still `6,7,8,9` before adding chunk.

Reply:
Good idea, I added that in https://github.com/prometheus/prometheus/pull/10051/commits/4d3d59494f79d57c0faa143668544adbedc81f97